content/en/docs/concepts/containers/images.md

```diff
@@ -380,6 +383,51 @@ Kubernetes supports specifying container image registry keys on a Pod.
 `imagePullSecrets` must all be in the same namespace as the Pod. The referenced
 Secrets must be of type `kubernetes.io/dockercfg` or `kubernetes.io/dockerconfigjson`.
 
+#### Ensure Image Pull Credential Verification {#ensureimagepullcredentialverification}
+
+{{< feature-state feature_gate_name="KubeletEnsureSecretPulledImages" >}}
+
+If the `KubeletEnsureSecretPulledImages` feature gate is enabled, Kubernetes
+will validate image credentials for every image that requires credentials 
+to be pulled, even if that image is already present on the node. This validation
+ensures that images in a pod request which have not been successfully pulled with the
+provided credentials must re-pull the images from the registry.
+Additionally, image pulls that re-use the same credentials which previously resulted 
+in a successful image pull will not need to re-pull from the registry and are instead 
+validated locally without accessing the registry.
+This is controlled by the`imagePullCredentialsVerificationPolicy` field in the
+[Kubelet configuration](/docs/reference/config-api/kubelet-config.v1beta1#ImagePullCredentialsVerificationPolicy).
+
+This configuration controls when image pull credentials must be verified if the
+image is already present on the node:
+
+ * `NeverVerify`: Mimics the behavior of having this feature gate disabled. If the image
+ is present locally, image pull credentials are not verified.
+ * `NeverVerifyPreloadedImages`: Images pulled outside the kubelet are not verified,
+ but all other images will have their credentials verified. This is the default behavior.
+ * `NeverVerifyAllowListedImages`: Images pulled outside the kubelet and mentioned within the 
+ `preloadedImagesVerificationAllowlist` specified in the kubelet config are not verified.
+ * `AlwaysVerify`: All images will have their credentials verified before they can
+ be used.
+
+With this feature enabled, image pulls which reuse the same credentials and image
+(in the case of a restarted container, for example), will not need to re-pull from
+the registry if that image is present either through caching or as a pre-pulled image.
+The kubelet verifies this by comparing the secret reference and a hash of the credentials used
+against the CRI image reference being requested. If the reference, hash, and image match a
+previously successful image pull, and a copy of the image is available locally either
+through pre-pulling or caching, then no call will be made to the registry and the local
+image will be used. Otherwise, the kubelet will use the provided credentials to re-pull
+the image from the registry before allowing the pod to run the image, even if the image is
+already present. This verification applies to [pre-pulled images](#pre-pulled-images),
+images pulled using node-wide secrets, and images pulled using pod-level secrets.
```

Comment:
I think this is a good description now, though I'd suggest dropping it as the intro to the section summarizes the higher level of the feature appropriately enough.